reflex/event.py

```diff
@@ -576,6 +581,27 @@ def download(url: str | Var, filename: Optional[str | Var] = None) -> EventSpec:
         if filename is None:
             filename = url.rpartition("/")[-1]
 
+    if filename is None:
```

Comment:
Can we add a couple comments in this function saying what the different url/encodings mean